src/pool/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) 2023 Intel Corporation
+# Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# libumf_pool_disjoint
+if(UMF_BUILD_LIBUMF_POOL_DISJOINT)
+    add_umf_library(disjoint_pool STATIC
+        pool_disjoint.cpp)
+
+    add_library(${PROJECT_NAME}::disjoint_pool ALIAS disjoint_pool)
```

Comment:
why don't you use add_umf_library? Or better - update this function and use it also for pool_jemalloc

Reply:
I don't use this custom wrapper of `add_library` as on aliased targets `target_include_directories` cannot be called. Beside that, note that `add_library(name ALIAS target)` is not the same function as ie. `add_library(name STATIC files)`. The former just creates an alias (another name) for an already existing CMake target whereas the latter creates a brand new CMake target.

I don't want to include aliasing into the `add_umf_library` function as well as we don't need all of our targets to have nice aliases with a namespace prefix, only those that are meant to be public and exposed to users when including the UMF repo with ie. FetchContent(). These I think should be limited to targets including headers and libraries containing the UMF API.

I believe that `pool_jemalloc` should be aliased, too. However, I save it for the more general UMF installation refactoring in another PR.

Reply:
ok thanks for the clarification. Regarding  aliasing pool_jemalloc - I will add a task about that